### PR TITLE
index dotcom

### LINKED ISSUE #377: Index Carbon for IBM.com

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #377

**Testing/Reviewing**

Run app on this branch with INDEX_ALL = 1 in .env.local and test out dotcom components: visuals, library, functions, punchout links. Leave any feedback here

### CODE REVIEW

**Review by mattrosno — CHANGES_REQUESTED** [review]
I see the React library has `externalDocsUrl: www.ibm.com/standards/web/carbon-for-ibm-dotcom`.

The React library detail page: http://localhost:3000/assets/ibmdotcom-react

Has a `External docs` link: http://localhost:3000/assets/www.ibm.com/standards/web/carbon-for-ibm-dotcom

We're probably looking for `http` to start a string to determine relative or absolute link? Let's fix in this PR?

File: services/web-app/data/libraries.js [hunk 1]
```diff
@@ -388,41 +388,46 @@ const devLibraries = {
   },
   'ibmdotcom-react': {
     host: 'github.com',
-    org: 'mattrosno',
+    org: 'francinelucca',
```
Comment (mattrosno) [inline services/web-app/data/libraries.js:391]: Once https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/8500 is merged, just a reminder to update the `org`s here to `carbon-design-system`, remove the `ref`, and update `prodLibraries` too.

**Review by mattrosno — CHANGES_REQUESTED** [review]
Can we add the `INDEX_ALL=1` env var to our test environment so we can see all in test? Or do we want to keep test and prod the same?

File: services/web-app/utils/string.js [hunk 2]
```diff
@@ -46,3 +46,13 @@ export const isJsonString = (str) => {
   }
   return true
 }
+
+/**
+ * Adds https protocol to url if no protocol present
+ * @param {string} url
+ * @returns {string} url with protocol
+ */
+export const getUrlWithProtocol = (url) => {
```
Comment (mattrosno) [inline services/web-app/utils/string.js:55]: @francinelucca do we need to `(url = "")` initialize with an empty string so if nothing gets passed in, the `startsWith()` doesn't throw an error?

Comment (francinelucca) [reply services/web-app/utils/string.js]: Done!

**Review by mattrosno — DISMISSED** [review]

**Review by mattrosno — DISMISSED** [review]

**Review by jharvey10 — DISMISSED** [review]

**Review by mattrosno — APPROVED** [review]

**Review by jharvey10 — APPROVED** [review]

**Pull request merged**